src/pudl/io_managers.py

```diff
@@ -412,12 +414,26 @@ def __init__(
                 connection opens a transaction to modify the database, it will be locked
                 until that transaction is committed.
         """
-        if not package:
+        if package is None:
             package = Package.from_resource_ids()
         self.package = package
         md = self.package.to_sql()
+        sqlite_path = Path(base_dir) / f"{db_name}.sqlite"
+        if not sqlite_path.exists():
+            raise RuntimeError(f"{sqlite_path} not initialized! Run pudl_reset_db.")
+
         super().__init__(base_dir, db_name, md, timeout)
 
+        existing_schema_context = MigrationContext.configure(self.engine.connect())
+        metadata_diff = compare_metadata(existing_schema_context, self.md)
+        if metadata_diff:
+            logger.info(f"Metadata diff:\n\n{metadata_diff}")
+            raise RuntimeError(
+                "Database schema has changed, try running `alembic revision "
+                "--autogenerate -m 'Your change message'` then `alembic "
+                "upgrade`."
+            )
```

Comment:
This error message should tell users to run `pudl_reset_db` right?

Reply:
Yes!